lms/static/js/inline_analytics.js

```diff
@@ -94,66 +97,55 @@ window.InlineAnalytics = (function() {
-                // currentIndex: number of rows already added to graphic
-                currentIndex = trs.length;
-                // valueIndex: index of the option for this row in the api result
-                valueIndex = valueId.replace('choice_', '');
-
-                // If the current row we are looking to add is less than the row indicated
-                // by the option for this index then add rows for those that are missing
-                if (currentIndex < valueIndex) {
-                    insertMissingRows(partId,
-                        currentIndex,
-                        valueIndex,
-                        correctResponse,
-                        choiceText,
-                        trs);
-                }
-
-                correct = result[index]['correct'];
-                count = result[index]['count'];
-                percent = Math.round(count * 1000 / (totalAttemptCount * 10));
-
-                if (correct) {
-                    answerClass = 'inline-analytics-correct';
+            	
+                // If the choice name is not in valueIdArray then we need to add a row
+                // for the missing data.
+                if (valueIdArray.indexOf(choiceNameArray[index]) === -1) {
+                	insertMissingRows(partId,
+                	    index,
+                	    index + 1,
+                	    correctResponse,
+                	    choiceText,
+                	    choiceNameArray,
+                	    trs);
                 } else {
-                    answerClass = 'inline-analytics-incorrect';
-                }
-                tr = $('<tr><td class="answer_box" title="' + choiceText[valueIndex] + '">' +
-                    (parseInt(valueIndex, 10) + 1) + '</td><td class="answer_box ' +
-                    answerClass + '"><span class="dot"></span></td><td class="answer_box">' +
-                    count + '</td><td class="answer_box">' + percent + '%</td></tr>');
-                trs.push(tr[0]);
-            }
+                    correct = result[index]['correct'];
+                    count = result[index]['count'];
+                    percent = Math.round(count * 1000 / (totalAttemptCount * 10));
 
-            // Generate rows for missing answers at the end of results
-            lastIndex = parseInt(result[result.length - 1]['value_id'].replace('choice_', ''), 10);
-            if (lastIndex < choiceText.length - 1) {
-                insertMissingRows(partId,
-                    lastIndex + 1,
-                    choiceText.length,
-                    correctResponse,
-                    choiceText,
-                    trs);
+                    if (correct) {
+                        answerClass = 'inline-analytics-correct';
+                    } else {
+                        answerClass = 'inline-analytics-incorrect';
+                    }
+                    tr = $('<tr><td class="answer_box" title="' + choiceText[index] + '">' +
```

Comment:
This string concatentation makes for pretty wonky code. How about embracing jQuery? Something along the lines of:

```javascript
tr = $('<tr>');
tdChoice = $('<td class="answer_box">');
tdChoice.attr('title', choiceText[index]);
tdChoice.text(parseInt(index, 10) + 1);
tdDot = $('<td class="answer_box">');
tdDot.addClass(answerClass);
tdDot.append($('<span class="dot">'));
tdPercent = $('<td class="answer_box">');
tdPercent.text(percent + '%');

tr.append(tdChoice);
tr.append(tdDot);
tr.append(tdPercent);
```